### PR TITLE
Sticky notification for websocket connection problems

### PR DESCRIPTION
**Pull request by hatsy**
## Overview
Creates a sticky notification when the websocket is not available, or when it is not authenticated (slightly different message). The notification is tied to the websocket status and not to the tricolor status (so there is no flicker on load). The websocket notification is cleared before a new one is added, so that there is always at most one websocket notification. 

### Testing connection issues:
- Run `make`
- Check that no notification is up
- Run `supervisorctl -c conf/supervisord_common.conf`
  - Run `stop websocket`
  - Check that there is a sticky notification
  - Run `start websocket`
  - Check that the notification is gone

### Testing authentication issues:
- Run `make`
- Run `supervisorctl -c conf/supervisord_common.conf`
- Edit `websocket_server.py` function `authenticate` to always return `AUTH_FAILED`
- restart the websocket server with `restart websocket`
- check that there is a sticky notification and the status is orange
- undo the changes and restart the websocket server
- check that the notification leaves (once the websocket server reloaded)

### CODE REVIEW

**Comment by hatsy** [comment]
Working on integrated the discussed changes.

**Comment by hatsy** [comment]
Updated the code and description, it's ready to go.

File: public/scripts/Notifications.jsx [hunk 1]
```diff
@@ -82,26 +94,39 @@ export function showNotification(note, type='info') {
...
   };
 }
 
 export function reducer(state={ notes: [] }, action) {
   switch (action.type) {
     case SHOW_NOTIFICATION: {
-      let { id, note, type } = action.payload;
+      let { id, note, type, unique, key } = action.payload;
+      var notes;
+      if (unique == true) {
```
Comment (stefanv) [inline public/scripts/Notifications.jsx]: if (unique)

**Review by stefanv — COMMENTED** [review]
As per our Slack discussion:

- key -> tag
- remove unique functionality, replace by something like `dispatch(clearNotificationByTag('websocket')).then(dispatch(showNotification(...)))`

**Comment by OrkoHunter** [comment]
@stefanv The @pep8speaks's comment on this PR is not justified. Apologies for that. I've raised an issue and will look into this as soon as possible.

File: public/scripts/WebSocket.jsx [hunk 2]
```diff
@@ -56,6 +57,16 @@ function getAuthToken(auth_url) {
...
 
+function clearNotifications(dispatch, tag) {
+  return new Promise(
+    (resolve, reject) => {
+      dispatch(hideNotificationByTag(tag));
+      resolve();
+    }
+  );
+}
+
```
Comment (hatsy) [inline public/scripts/WebSocket.jsx]: @stefanv I was considering moving this function to Notifications.jsx what do you think?

Comment (stefanv) [reply public/scripts/WebSocket.jsx]: Agreed that's a better place for it.

Comment (stefanv) [reply public/scripts/WebSocket.jsx]: Although, I'm not quite sure this is necessary.

**Comment by stefanv** [comment]
Beautiful!  Thank you, @hatsy, this is an excellent contribution.

**Review by stefanv — APPROVED** [review]

**Comment by stefanv** [comment]
@acrellin Do you mind reviewing?

**Review by acrellin — APPROVED** [review]

**Pull request merged**